server/src/main/java/io/druid/server/http/RulesResource.java

```diff
@@ -79,9 +89,40 @@ public Response setDatasourceRules(
       final List<Rule> rules
   )
   {
-    if (databaseRuleManager.overrideRule(dataSourceName, rules)) {
+    if (databaseRuleManager.overrideRule(
+        dataSourceName,
+        new RulesConfig(rules, new AuditInfo("NULL", "NULL", "NULL"))
+    )) {
       return Response.ok().build();
     }
     return Response.status(Response.Status.INTERNAL_SERVER_ERROR).build();
   }
+
+  @POST
+  @Path("/{dataSourceName}/rulesConfig")
```

Comment:
do we really need another endpoint? I like rainer's approach to passes X-Rainer-Author, X-Rainer-Comment as headers. That makes it possible to add audit info to any existing endpoint without requiring additional serde.
Given that we may want to add those to other endpoints in the future, I feel this would be more flexible.